### ISSUE TITLE
[BUG] choppy messed up audio linux

### ISSUE THREAD

**Issue by ballerburg9005** [ZDoom/gzdoom#1730]
### GZDoom version

GZDoom g4.9pre-214-gba83879e9-m - 2022-08-30 22:21:08 +0100 - SDL version

### Which game are you running with GZDoom?

Doom

### What Operating System are you using?

Linux x86_64

### Please describe your specific OS version

Archlinux alsa (without pulseaudio)

### Relevant hardware info

emu10k driver (old Sound Blaster Audigy 2)

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

The sound is messed up as per provided sample: https://vocaroo.com/1iXNooJWY3Ks (It sounds like every few hundred milliseconds the sound is cutting out for 2ms, virtually identical to how it sounds if the buffer size is too small in DOSBox.)

At first this looked like a rare but very well known problem with the emu10k driver that has an established cure, which is to set up a resampler dmix in asoundrc to increase period_size, buffer_size and use rate 44100 instead of the native rate of 48000 which as I understand somehow is responsible for the bug sometimes. However in the case of gzdoom this has no effect, which suggests that this is a new problem unrelated to the hardware. 

I found internet threads that suggest for Windows to delete the OpenAL.dll file, however I fail to see how something similar could be accomplished in Linux. I had a brief look at the code for the undocumented variables such as snd_backend or snd_aldevice, but it appears to me that they only take "default" and "null" as values. SImilarly none of the advanced audio settings such as HRTF or snd_buffersize had any effect.

The bug affects all sound, not just music insofar as that it is only emulated MIDI.

### Your configuration

```ini
I tried all sorts of combinations of snd_ settings / advanced settings, none had any effect whatsoever.
```

### Provide a Log

GZDoom g4.9pre-214-gba83879e9-m - 2022-08-30 22:21:08 +0100 - SDL version
Compiled on Sep  3 2022

OS: Arch Linux, Linux 5.18.7-arch1-1 on x86_64
GZDoom version g4.9pre-214-gba83879e9-m
W_Init: Init WADfiles.
 adding /usr/lib/gzdoom/gzdoom.pk3, 669 lumps
 adding /usr/lib/gzdoom/game_support.pk3, 2511 lumps
 adding /mnt/2/GAMES/doom1666/doom1.wad, 1264 lumps
S_Init: Setting up sound.
I_InitSound: Initializing OpenAL
  Opened device ALSA Default
  EFX enabled
I_Init: Setting up machine state.
CPU speed: 3311 MHz
CPU Vendor ID: GenuineIntel
  Name: Intel(R) Core(TM) i5-2500 CPU @ 3.30GHz
  Family 6, Model 42, Stepping 7
  Features: SSE2 SSE3 SSSE3 SSE4.1 SSE4.2 AVX HyperThreading
V_Init: allocate screen.
ST_Init: Init startup screen.
Checking cmd-line parameters...
S_InitData: Load sound definitions.
G_ParseMapInfo: Load map definitions.
Texman.Init: Init texture manager.
ParseTeamInfo: Load team definitions.
LoadActors: Load actor definitions.
script parsing took 153.38 ms
R_Init: Init Doom refresh subsystem.
DecalLibrary: Load decals.
M_Init: Init menus.
P_Init: Init Playloop state.
ParseSBarInfo: Loading custom status bar definition.
D_CheckNetGame: Checking network game status.
player 1 of 1 (1 nodes)
Using video driver x11
GL_VENDOR: NVIDIA Corporation
GL_RENDERER: NVIDIA GeForce GTX 660/PCIe/SSE2
GL_VERSION: 4.6.0 NVIDIA 470.129.06 (Core profile)
GL_SHADING_LANGUAGE_VERSION: 4.60 NVIDIA

Max. texture size: 16384
Max. texture units: 32
Max. varying: 124
Max. combined shader storage blocks: 96
Max. vertex shader storage blocks: 16
Resolution: 356 x 200

----------------------------------------

E1M3 - Toxin Refinery

Line 933's right edge is unconnected
Ballerburg mutated.

**Comment by coelckers** [comment]
There's not much we can do on our side here. The sound is entirely driven by OpenAL Soft so you may have to sort it out with its maintainer.

**Comment by kcat** [comment]
> At first this looked like a rare but very well known problem with the emu10k driver that has an established cure, which is to set up a resampler dmix in asoundrc to increase period_size, buffer_size and use rate 44100 instead of the native rate of 48000 which as I understand somehow is responsible for the bug sometimes.

How did you do this, and did you set up this dmix configuration to be the default ALSA device? If you just create a new dmix device but don't tell anything to use it, it of course won't have any effect.

**Comment by ballerburg9005** [comment]
Thanks for the info. I tried Supertuxracer for comparison, which also uses OpenAL, and the sound is also messed up in a very similar (but not quite identical) way. 

Turning the rate to 8000 in the .asoundrc gets rid of the crackling in Gzdoom, but naturally sound quality will be very poor at this rate.

However I observed several odd things for Gzdoom:

1. Changing the rate inside Gzdoom doesn't seem to make any audible difference, which is totally unlike changing the rate in the .asoundrc dmixer, which sounds exactly like you would expect the lower quality to sound. And .asoundrc removes the crackling at rate 8000 while Gzdoom setting does nothing.
2. Changing the buffer via snd_buffersize in Gzdoom as well as changing the buffer in .asoundrc outside the game doesn't result in any audible difference such as a different crackling interval (but it should, like it usually does like for Supertuxracer). If the buffer is too low, the sound should stop working entirely, which it does for .asoundrc but not for snd_buffersize from Gzdoom settings. This suggests to me that the snd_buffersize setting is not working, and possibly that OpenAL is set up by Gzdoom to override the buffer settings from the hardware.

If your code was setting up OpenAL with fixed buffer/rate/period? parameters that are unaffected by the user settings, rather than letting the hardware decide, this could be the source of the problem, rather than OpenAL itself. 

However as Supertuxracer also uses OpenAL and has a very similar issue, it doesn't look too promising that this would be really the only cause of the problem. I will try to do more testing.
